components/director/internal/domain/oauth20/resolver.go

```diff
@@ -0,0 +1,142 @@
+type ApplicationService interface {
+	Exist(ctx context.Context, id string) (bool, error)
+}
+
+//go:generate mockery -name=RuntimeService -output=automock -outpkg=automock -case=underscore
+type RuntimeService interface {
+	Exist(ctx context.Context, id string) (bool, error)
+}
+
+//go:generate mockery -name=IntegrationSystemService -output=automock -outpkg=automock -case=underscore
+type IntegrationSystemService interface {
+	Exist(ctx context.Context, id string) (bool, error)
+}
+
+//go:generate mockery -name=SystemAuthConverter -output=automock -outpkg=automock -case=underscore
+type SystemAuthConverter interface {
+	ToGraphQL(model *model.SystemAuth) *graphql.SystemAuth
+}
+
+//go:generate mockery -name=Service -output=automock -outpkg=automock -case=underscore
+type Service interface {
+	CreateClient(ctx context.Context, objectType model.SystemAuthReferenceObjectType) (*model.OAuthCredentialDataInput, error)
+	DeleteClient(ctx context.Context, clientID string) error
+}
+
+type Resolver struct {
+	transact       persistence.Transactioner
+	svc            Service
+	systemAuthSvc  SystemAuthService
+	systemAuthConv SystemAuthConverter
+	appSvc         ApplicationService
+	rtmSvc         RuntimeService
+	isSvc          IntegrationSystemService
+}
+
+func NewResolver(transactioner persistence.Transactioner, svc Service, appSvc ApplicationService, rtmSvc RuntimeService, isSvc IntegrationSystemService, systemAuthSvc SystemAuthService, systemAuthConv SystemAuthConverter) *Resolver {
+	return &Resolver{transact: transactioner, svc: svc, appSvc: appSvc, rtmSvc: rtmSvc, systemAuthSvc: systemAuthSvc, isSvc: isSvc, systemAuthConv: systemAuthConv}
+}
+
+func (r *Resolver) GenerateClientCredentialsForRuntime(ctx context.Context, id string) (*graphql.SystemAuth, error) {
+	return r.generateClientCredentials(ctx, model.RuntimeReference, id)
+}
+
+func (r *Resolver) GenerateClientCredentialsForApplication(ctx context.Context, id string) (*graphql.SystemAuth, error) {
+	return r.generateClientCredentials(ctx, model.ApplicationReference, id)
+}
+
+func (r *Resolver) GenerateClientCredentialsForIntegrationSystem(ctx context.Context, id string) (*graphql.SystemAuth, error) {
+	return r.generateClientCredentials(ctx, model.IntegrationSystemReference, id)
+}
+
+func (r *Resolver) generateClientCredentials(ctx context.Context, objType model.SystemAuthReferenceObjectType, objID string) (*graphql.SystemAuth, error) {
+	tx, err := r.transact.Begin()
+	if err != nil {
+		return nil, err
+	}
+	defer r.transact.RollbackUnlessCommited(tx)
+
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	exists, err := r.checkObjectExist(ctx, objType, objID)
+	if err != nil {
+		return nil, errors.Wrap(err, "while checking if runtime exists")
+	}
+	if !exists {
+		return nil, fmt.Errorf("%s with ID '%s' not found", objType, objID)
+	}
+
+	clientCreds, err := r.svc.CreateClient(ctx, objType)
+	if err != nil {
+		return nil, err
+	}
+	if clientCreds == nil {
```

Comment:
In theory this nil here shouldn't happen anytime, because service handles it well, but this is additional check just in case of further changes in service. I'm not feeling well to assume some pointer will be always not nill.